projects/iot-cloud/software/lora2mqtt/src/utils/functions.py

```diff
@@ -0,0 +1,20 @@
+from utils.exceptions import RestException
+
+
+def resolve_mapping(mapping_config, app_id, dev_id, auth_token):
+    """
+    Search the given device in the mapping are returns the key that should be used for publishing data to MQTT.
+    If any requirement is not fit, when it returns a RestException.
+    """
+    app_dict = mapping_config.get(app_id)
```

Comment:
This will throw if the key doesn't exist. Use a default `None` value if you want to follow this pattern.

Reply:
I did not validated the config file, but I got the assumption that it is always valid.
That is a component (config server) to component (this component) communication, so there is no reason to raise REST errors for it. It will lead to a 500 HTTP error on a subsequent line, which is not wrong, in my opinion.